src/node_osrm_support.hpp

```diff
@@ -733,6 +733,74 @@ argumentsToTripParameter(const Nan::FunctionCallbackInfo<v8::Value> &args,
         return trip_parameters_ptr();
     }
 
+    if (obj->Has(Nan::New("roundtrip").ToLocalChecked()))
+    {
+        auto roundtrip = obj->Get(Nan::New("roundtrip").ToLocalChecked());
+        if (roundtrip->IsBoolean())
+        {
+            params->roundtrip = roundtrip->BooleanValue();
+        }
+        else
+        {
+            Nan::ThrowError("'roundtrip' param must be a boolean");
+            return trip_parameters_ptr();
+        }
+    }
+
+    if (obj->Has(Nan::New("source").ToLocalChecked()))
+    {
+        v8::Local<v8::Value> source = obj->Get(Nan::New("source").ToLocalChecked());
+
+        if (!source->IsString())
+        {
+            Nan::ThrowError("Source must be a string: [any, first]");
+            return trip_parameters_ptr();
+        }
+
+        std::string source_str = *v8::String::Utf8Value(source);
```

Comment:
Again I tried to implement this but didn't manage to get it to work, but I'll leave this for now, as the pattern here is different from elsewhere in the rest of the file. This can be addressed in a refactor PR that changes the rest of the file as well.